packages/web3-core-requestmanager/src/index.js

```diff
@@ -123,11 +123,27 @@ RequestManager.prototype.setProvider = function (provider, net) {
             });
         });
 
-        // notify all subscriptions about the close condition
+        // notify all subscriptions about bad close conditions
         this.provider.on('close', function close(event) {
-            _this.subscriptions.forEach(function (subscription) {
-                subscription.callback(new Error('CONNECTION ERROR: The connection got closed with the close code `' + event.code + '` and the following reason string `' + event.reason + '`'));
-            });
+            // The node.net module emits ('close', isException)
+            const ipcCloseError = typeof event === 'boolean' && event;
```

Comment:
This could be moved out to a private helper function on the ``RequestManager`` object. Such a function would make this un-clean inline comment obsolete. :)